tests/test_exporter.py

```diff
@@ -1130,6 +1130,43 @@ def test_exporter_can_export_requirements_txt_with_nested_packages_cyclic(
     assert content == expected
 
 
+def test_exporter_can_export_requirements_txt_with_circular_dependency(
+    tmp_path: Path, poetry: Poetry
+) -> None:
+    poetry.locker.mock_lock_data(  # type: ignore[attr-defined]
+        {
+            "package": [
+                {
+                    "name": "foo",
+                    "version": "1.2.3",
+                    "category": "main",
+                    "optional": False,
+                    "python-versions": "*",
+                    "dependencies": {"my-package": {"version": "1.2.3"}},
```

Comment:
```suggestion
                    "dependencies": {poetry.package.pretty_name: {"version": "1.2.3"}},
```

This way, the risk of accidentally making the test useless is reduced.